pulp_ansible/app/serializers.py

```diff
@@ -270,17 +271,65 @@ class CollectionVersionSerializer(SingleArtifactContentSerializer, ContentChecks
         help_text=_("The namespace of the collection."), max_length=32
     )
 
-    repository = serializers.URLField(
-        help_text=_("The URL of the originating SCM repository."), allow_blank=True, max_length=128
-    )
-
     tags = TagNestedSerializer(many=True, read_only=True)
 
     version = serializers.CharField(help_text=_("The version of the collection."), max_length=32)
 
+    def deferred_validate(self, data):
+        """
+        Validates data.
+
+        Args:
+            data (dict): User data to validate
+
+        Returns:
+            dict: Validated data
+
+        Raises:
+            rest_framework.serializers.ValidationError: If invalid data
+
+        """
+        data = super().deferred_validate(data)
+        artifact = data["artifact"]
+        with artifact.file.open() as artifact_file:
+            importer_result = import_collection(artifact_file)
+
+        collection_info = importer_result["metadata"]
```

Comment:
I think the `defered_validate()` should stop here and move the functionality below to a `create()` call on the `CollectionVersionSerializer`. Otherwise it's [validated and created at once in this call](https://github.com/pulp/pulpcore/pull/305/files#diff-96bbb4ed1da9a24fe45e8f0626df8ad4R17), and then it will be "created" again later with its own `save()` [here](https://github.com/pulp/pulpcore/pull/305/files#diff-96bbb4ed1da9a24fe45e8f0626df8ad4R18).